content/js1/sprints/4/prep/index.md

```diff
@@ -8,4 +8,666 @@ backlog= 'Module-JS1'
+- Explain the process involved in looking up how to write tests for someone using a language other than JavaScript
+- Express an acceptance criterion as a test case
+
+## Notes
+
+### 🧮 Ordinal numbers
+
+Let's imagine you're working in a 10 storey office building - so there are 10 different levels.
+We need a way to describe each level of the building.
+
+We start on the **ground floor** of the building - level with the ground.
+
+We can use an **ordinal number** to describe the other levels in the building.
+
+Up from the ground floor, we are then on the 1**st** floor (fir**st** floor)
+Up from the **1st floor**, we are on the 2**nd** floor (seco**nd** floor)
+
+To form the ordinal number we take a number and add the correct **suffix**:
+
+> number + **suffix** -> ordinal number
+> e.g.
+> 1 + '**st**' -> '1st'
+> 2 + '**nd**' -> '2nd'
+
+The suffix comes from the word used to describe each number, like fir**st**, seco**nd**, thi**rd** etc.
+
+Here is a list of the first 10 ordinal numbers.
+
+| number | ordinal number |
+| ------ | -------------- |
+| 1      | 1**st**        |
+| 2      | 2**nd**        |
+| 3      | 3**rd**        |
+| 4      | 4**th**        |
+| 5      | 5**th**        |
+| 6      | 6**th**        |
+| 7      | 7**th**        |
+| 8      | 8**th**        |
+| 9      | 9**th**        |
+| 10     | 10**th**       |
+
+{{<note type="exercise" title="Exercise">}}
+
+Using the information from above, try to answer the following questions:
+
+a) What will the ordinal number be for 21?
+b) What will the ordinal number be for 40?
+c) What will the ordinal number be for 49?
+d) What will the ordinal number be for 13?
+
+{{</note>}}
+
+{{<note type="exercise" title="Exercise">}}
+
+### 📆 Using dates
+
+Use ordinal numbers to write the dates of the following events:
+
+a) Tomorrow
+b) A week from now
+c) Easter sunday 2024
+d) When is Eid expected to occur in 2024
+
+{{</note>}}
+
+Let's consider a function called `getOrdinalNumber` that needs to work as follows:
+
+- it takes one argument - a whole number, like 1, 2, 3, etc
+- returns a _string_ that represents the ordinal number
+
+E.g.
+
+```js
+getOrdinalNumber(1); // returns "1st";
+getOrdinalNumber(2); // returns "2nd";
+getOrdinalNumber(6); // returns "6th";
+```
+
+The requirements above form a specification of how the function `getOrdinalNumber` should behave.
+
+{{<note type="definition" title="definition">}}
+A **specification** is a description of how a piece of software works.
+In other words, a specification is a set of requirements for how a piece of software should behave.
+{{</note>}}
+
+Once we have a specification for how the function should work we can create many cases showing how we expect the function `getOrdinalNumber` to behave when it is called with certain inputs.
+
+### 🖼️ Using a test framework
+
+To implement a function like `getOrdinalNumber` we can try specifying a single case:
```

Comment:
I'm not sure "To implement" is a great start to this sentence, because this _isn't_ implementing the function. It's using it.

Maybe "To help us think about the requirements of a function like `getOrdinalNumber`"?

Reply:
yeah definitely